fix: point-in-time APIs

pwnb0t opened the pull request:
### Description
Reverted https://github.com/opensearch-project/opensearch-php/pull/139
Fix the Point-in-Time APIs -- the URIs just needed adjustment.

Tested it locally in my project and it is now working properly for me.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-php/issues/141
Related: https://github.com/opensearch-project/opensearch-php/issues/137 these files were removed, but just needed to be fixed instead

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

shyim reviewed (dismissed):
thankss!

harshavamsi commented:
Hi @pwnb0t the core logic looks good to me. But can we add tests to ensure that this actually works? Can we also update the `USER_GUIDE.md` to show these new endpoints? Thanks!

shyim commented:
The tests seems to fail

pwnb0t commented:
@VachaShah I have changed them to `createPointInTime` and `deletePointInTime` to better match the API. Let me know if you wanted something different than that.

@harshavamsi I've added tests and also added to the `USER_GUIDE.md`. I wasn't sure how detailed the example I should put in the USER_GUIDE, but I think I've captured the main points of using a PIT. Creating the PIT and then using it to page through results, as well as showing how to do the `search_after` part. Please let me know if you would like refinement to that.

@shyim and others, I have failures in these new tests and I'm not fully certain on what needs to be done. Some of the tests are running older versions of OpenSearch. The PIT APIs were introduced in OpenSearch 2.4 according to the documentation, but I only see these APIs in the 2.5 and above documentation (https://opensearch.org/docs/2.5/search-plugins/point-in-time/). Is there a way to exclude the PIT tests from unsupported versions?

shyim commented:
@pwnb0t the best would be to do an $client->info in the test setUp and use static::markTestSkipped if the version is too low :)

pwnb0t commented:
@shyim I've added that check for the version. I also added opensearch 2.5.0 to the test matrix -- let me know if I should back that out.

Additionally, I did test 2.4.1 locally just to ensure that the PIT APIs work since they did not appear in the documentation until 2.5.0.

shyim commented on .github/workflows/test.yml:
```diff
@@ -145,6 +145,7 @@ jobs:
           - opensearchproject/opensearch:1.3.6
           - opensearchproject/opensearch:2.1.0
           - opensearchproject/opensearch:2.2.1
+          - opensearchproject/opensearch:2.5.0
```
Can you remove the elasticsearch-oss?

pwnb0t replied:
Yes, I'll add a check on `version.distribution == opensearch`

pwnb0t commented:
@shyim I split out the version check into `isOpenSearchVersionAtLeast` and `isElasticSearchVersionAtLeast` then applied the `isElasticSearchVersionAtLeast` to the existing `getVersion` usages and used `isOpenSearchVersionAtLeast` in the PIT tests. Hopefully that will satisfy the requirements.

pwnb0t commented:
I think I've got it all fixed up now. I fixed the version checks as well as changed the existing ones a bit (so now it reads like ">= 7.4.0" rather than "> 7.3.99").

Could you rerun the tests on this?

shyim reviewed (approved):
Looks good. Thanks for your PR! ❤️

VachaShah reviewed (approved):
LGTM. Thank you @pwnb0t!
